sia/src/app_client.rs

```diff
@@ -0,0 +1,235 @@
+            url: url.trim_end_matches('/').to_string(),
+            password,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn slab(&self, slab_id: &SlabID) -> Result<Slab> {
+        self.get_json(&format!("/slab/{slab_id}")).await
+    }
+
+    #[allow(dead_code)]
+    pub async fn pin_slab(&self, slab: &SlabPinParams) -> Result<SlabID> {
+        self.post_json("/slabs", serde_json::to_string(&slab)?)
+            .await
+    }
+
+    /// Helper to send a GET request with basic auth and parse the JSON
+    /// response.
+    async fn get_json<D: DeserializeOwned>(&self, path: &str) -> Result<D> {
+        Ok(self
+            .client
+            .get(format!("{}{}", self.url, path))
+            .basic_auth("", self.password.clone())
+            .send()
+            .await?
+            .json::<D>()
+            .await?)
+    }
+
+    // Helper to send a POST request with basic auth and parse the JSON
+    // response.
+    async fn post_json<B: Into<Body>, D: DeserializeOwned>(
+        &self,
+        path: &str,
+        body: B,
+    ) -> Result<D> {
+        Ok(self
+            .client
+            .post(format!("{}{}", self.url, path))
+            .basic_auth("", self.password.clone())
+            .body(body)
+            .send()
+            .await?
+            .json::<D>()
+            .await?)
+    }
+}
+
+impl Drop for Client {
+    fn drop(&mut self) {
+        if let Some(password) = &mut self.password {
+            password.zeroize();
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use httptest::http::Response;
+    use httptest::matchers::*;
+    use httptest::{Expectation, Server};
+
+    #[tokio::test]
+    async fn test_basic_auth() {
+        let server = Server::run();
+
+        // expect 1 authenticated get and 1 authenticated post request
+        server.expect(
+            Expectation::matching(request::headers(contains((
+                "authorization",
+                "Basic OnBhc3N3b3Jk",
+            ))))
+            .times(2)
+            .respond_with(Response::builder().status(200).body("{}").unwrap()),
+        );
+
+        let client = Client::new(server.url("/").to_string(), Some("password".to_string()));
+        let _: Result<()> = client.get_json("/").await;
+        let _: Result<()> = client.post_json("/", "{}").await;
+    }
+
+    #[tokio::test]
+    async fn test_slab() {
+        let slab = Slab {
+            id: "43e424e1fc0e8b4fab0b49721d3ccb73fe1d09eef38227d9915beee623785f28"
+                .parse()
+                .unwrap(),
+            encryption_key: [
+                186, 153, 179, 170, 159, 95, 101, 177, 15, 130, 58, 19, 138, 144, 9, 91, 181, 119,
+                38, 225, 209, 47, 149, 22, 157, 210, 16, 232, 10, 151, 186, 160,
+            ],
+            min_shards: 1,
+            sectors: vec![Sector {
+                root: "826af7ab6471d01f4a912903a9dc23d59cff3b151059fa25615322bbf41634d6"
```

Comment:
`hash_256!()`

You know, I added really nice compile time macros and you continue to not use them.